examples/v1/firmware/firmware.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/bmc-toolbox/bmclib"
+	"github.com/bmc-toolbox/bmclib/devices"
+	"github.com/bombsimon/logrusr/v2"
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+
+	// set BMC parameters here
+	host := ""
+	port := ""
+	user := ""
+	pass := ""
+
+	l := logrus.New()
+	l.Level = logrus.TraceLevel
+	logger := logrusr.New(l)
+
+	if host == "" || user == "" || pass == "" {
+		log.Fatal("required host/user/pass parameters not defined")
+	}
+
+	cl := bmclib.NewClient(host, port, user, pass, bmclib.WithLogger(logger))
+
+	err := cl.Open(ctx)
+	if err != nil {
+		log.Fatal(err, "bmc login failed")
+	}
+
+	defer cl.Close(ctx)
+
+	for _, update := range []string{"/tmp/E6D4INL2.09C.ima"} {
+		fh, err := os.Open(update)
+		if err != nil {
+			l.Fatal(err)
+		}
+
+		taskID, err := cl.FirmwareInstall(ctx, devices.SlugBMC, devices.FirmwareApplyOnReset, true, fh)
+		if err != nil {
+			l.Fatal(err)
+		}
+
+		state, err := cl.FirmwareInstallStatus(ctx, "", taskID, "5.00.00.00")
```

Comment:
The argument order looks wrong here

Reply:
thanks, fixed in 91fc9048e4ee0e6a0ec524493cb0bd93153e87f5